README.md

```diff
@@ -3,23 +3,32 @@
 [![Codacy Badge](https://api.codacy.com/project/badge/Grade/a877b7395f2d475aa79c08daf665dc3c)](https://www.codacy.com/app/dpgraham/appium-espresso-driver?utm_source=github.com&amp;utm_medium=referral&amp;utm_content=appium/appium-espresso-driver&amp;utm_campaign=Badge_Grade)
 [![Greenkeeper badge](https://badges.greenkeeper.io/appium/appium-espresso-driver.svg)](https://greenkeeper.io/)
 
-Espresso integration for Appium
+Appium Espresso Driver is a test automation framework for Android that uses [Espresso](https://developer.android.com/training/testing/espresso/) as the underlying test technology. Appium Espresso Driver is part of the Appium framework.
 
-Note: this is an experimental work in progress and is not ready for use.
+## Comparison with UiAutomator2
 
-## Developers
+The key difference between [UiAutomator2 Driver](https://github.com/appium/appium-uiautomator2-driver) and Espresso Driver is that UiAutomator2 is a black-box testing framework, and Espresso is a "grey-box" testing framework. The Espresso Driver itself is black-box (no internals of the code are exposed to the tester), but the Espresso framework itself has access to the internals of Android applications. This distinction has a few notable benefits. It can find elements that aren't rendered on the screen, it can identify elements by the Android View Tag and it makes use of [IdlingResource](https://developer.android.com/reference/android/support/test/espresso/IdlingResource) which blocks the framework from running commands until the UI thread is free. The downside to this is that you can only automate the app-under-test (AUT)
 
-### Building Espresso Server
-```
-npm run build:server
-```
-or
-```
-cd espresso-server
-./gradlew clean assembleDebug assembleAndroidTest
-```
+## Troubleshooting
 
-### Layout
+* If there are ever problems starting a session, try setting the capability `forceEspressoRebuild=true` and retrying. This will rebuild a fresh Espresso Server APK. If the session is succcesful, set it back to false so that it doesn't re-install on every single test.
```

Comment:
We should also mention signing here. that is is required that both the server and the are signed with the same debug singature